src/shared/schemas.ts

```diff
@@ -251,6 +245,93 @@ export async function updateSchemaFromRemote(params: {
     }
 }
 
+/**
+ * Fetches url content and caches locally. Uses E-Tag to determine if cached
+ * content can be used.
+ * @param params.destination Path to local file
+ * @param params.eTag E-Tag to send with fetch request. If this matches the url's it means we can use our cache.
+ * @param params.url Url to fetch from
+ * @param params.cacheKey Cache key to check version against
+ * @param params.extensionContext VSCode extension context
+ */
+export async function updateSchemaFromRemoteETag(params: {
+    destination: vscode.Uri
+    eTag?: string
+    url: string
+    cacheKey: string
+    extensionContext: vscode.ExtensionContext
+    title: string
+}): Promise<void> {
+    const cachedETag = params.extensionContext.globalState.get<string>(params.cacheKey)
+
+    // Check that the cached file actually can be fetched. Else we might
+    // never update the cache.
+    const fileFetcher = new FileResourceFetcher(params.destination.fsPath)
+    const cachedContent = await fileFetcher.get()
+
+    // Determine if existing cached content is sufficient
+    const needsUpdate = cachedETag === undefined || cachedETag !== params.eTag
+    const hasCachedContent = cachedContent !== undefined
+    if (hasCachedContent && !needsUpdate) {
+        return
+    }
+
+    try {
+        // Only use our cached E-Tag if we have it + cached content
+        const eTagToRequest = !!cachedETag && cachedContent !== undefined ? cachedETag : params.eTag
+
+        const httpFetcher = new HttpResourceFetcher(params.url, { showUrl: true })
+        const response = await httpFetcher.getNewETagContent(eTagToRequest)
+        if (!response) {
+            throw new Error(`failed to resolve schema: ${params.destination}`)
+        }
+
+        const { content, eTag: latestETag } = response
+        if (content === undefined) {
+            // Our cached content is the latest
+            return
+        }
+
+        params.eTag = latestETag
+        doCacheContent(content, params)
```

Comment:
I would log something because this isn't being awaited:
```suggestion
        doCacheContent(content, params).catch(err => getLogger().warn('schemas: failed to cache result: %s', err))
```

Reply:
I didn't realize I missed the await, thanks for catching that. Have we thought about enabling a lint warning when we don't use an `await`?

Reply:
There's a PR out there somewhere that turns on the rule. Not using `await` is _sometimes_ intentional so adding the rule is not as straightforward as one might hope.